src/objects/zcl_abapgit_objects_super.clas.abap

```diff
@@ -141,28 +142,97 @@ CLASS ZCL_ABAPGIT_OBJECTS_SUPER IMPLEMENTATION.
 
   METHOD corr_insert.
 
-    DATA: ls_object TYPE ddenqs.
+    DATA: lv_object           TYPE string,
+          lv_object_class     TYPE string,
+          lv_gui_is_available TYPE c,
+          lv_korrnum_check    TYPE e070-trkorr,
+          lv_ordernum_check   TYPE e070-trkorr,
+          lv_korrnum          TYPE e070-trkorr.
+
+    IF iv_object_class IS NOT INITIAL.
+      lv_object_class = iv_object_class.
+      IF iv_object_class = 'DICT'.
+        CONCATENATE ms_item-obj_type ms_item-obj_name INTO lv_object.
+      ELSE.
+        lv_object = ms_item-obj_name.
+      ENDIF.
+    ELSE.
+      lv_object_class = ms_item-obj_type.
+      lv_object       = ms_item-obj_name.
+    ENDIF.
 
+    CALL FUNCTION 'GUI_IS_AVAILABLE'
+      IMPORTING
+        return = lv_gui_is_available.
 
-    ls_object-objtype = ms_item-obj_type.
-    ls_object-objname = ms_item-obj_name.
+    IF lv_gui_is_available EQ abap_true.
+      CALL FUNCTION 'RS_CORR_INSERT'
+        EXPORTING
+          object              = lv_object
+          object_class        = lv_object_class
+          devclass            = iv_package
+          master_language     = mv_language
+          global_lock         = abap_true
+          mode                = 'I'
+        EXCEPTIONS
+          cancelled           = 1
+          permission_failure  = 2
+          unknown_objectclass = 3
+          OTHERS              = 4.
+    ELSE.
+
+      "check whether object is currently locked in different transport
+      CALL FUNCTION 'RS_CORR_CHECK'
```

Comment:
Looks like this part relates to https://github.com/larshp/abapGit/issues/2345 ? and not issue #2378 directly

Suggest removing this fix from this PR, and submitting it separately. I also wonder why the logic is different for background and foreground mode?

Reply:
In foreground mode, a popup is shown that the object is locked in a different request and the object can only be recorded in that request. In background mode, the object is automatically assigned to the other request w/o showing a popup. I agree, that this is a different issue than #2378 - I've added the complete `CORR_INSERT` method (with the 'non SAPGUI version') as this is required for our project. If this is too much for #2378, I'll reduce the coding in this pull request and deal with the difference in our project or create a different pull request later...

Reply:
the transport popup is handled via `zcl_abapgit_default_transport` and `zcl_abapgit_ui_factory=>get_popups( )->popup_transport_request`, in GUI_DESERIALIZE. If this method shows a popup, then something is wrong somewhere else, so this looks like a workaround for a broken feature that is implemented elsewhere

Reply:
how can above issue be reproduced?